mitodl/mit-open
# Add a Search Page

### What are the relevant tickets?
Closes #490 

### Description (What does it do?)
This PR adds a search page. We don't have final designs for this, but I used https://www.figma.com/proto/bRb19HaucO5loHGSU81nFG/MIT_Prototype?page-id=0%3A1&type=design&node-id=9-6102&viewport=4023%2C1145%2C0.25&t=6VyDzTtHVzsZ3Nyv-1&scaling=scale-down&starting-point-node-id=48%3A2446&show-proto-sidebar=1 as inspiration.

**Note:** After discussing with @Ferdi, only "Topics" facet is supported for now (and "Resource Type" via the tabs).

### Screenshots (if appropriate):

![Screenshot 2024-03-18 at 11 20 17 AM](https://github.com/mitodl/mit-open/assets/9010790/09a697c2-889f-4f55-ae5a-272163472666)


### How can this be tested?
1. Ensure you have a working search index https://github.com/mitodl/mit-open?tab=readme-ov-file#enabling-searching-the-course-catalog-on-opensearch
2. Visit http://localhost:8063/; enter something in the searchbox and hit enter.
3. You should now be on the searchpage. Try facets in sidebar, tabs for resource_type facet, query text, and pagination.
    - results should update appropriately
    - URL should update appropriately
